test/new.jl

```diff
@@ -0,0 +1,1817 @@
+module NewTests
+
+using  Test, UUIDs
+import Pkg, LibGit2
+using  Pkg.Types: PkgError
+using  Pkg.Resolve: ResolverError
+using  ..Utils
+
+Pkg.DEFAULT_IO[] = IOBuffer()
+
+general_uuid = UUID("23338594-aafe-5451-b93e-139f81909106") # UUID for `General`
+exuuid = UUID("7876af07-990d-54b4-ab0e-23690620f79a") # UUID for `Example.jl`
+json_uuid = UUID("682c06a0-de6a-54ab-a142-c8b1cf79cde6")
+markdown_uuid = UUID("d6f4376e-aef5-505a-96c1-9c027394607a")
+test_stdlib_uuid = UUID("8dfed614-e22c-5e08-85e1-65c5234f0b40")
+unicode_uuid = UUID("4ec0a83e-493e-50e2-b9ac-8f72acf5a8f5")
+unregistered_uuid = UUID("dcb67f36-efa0-11e8-0cef-2fc465ed98ae")
+simple_package_uuid = UUID("fc6b7c0f-8a2f-4256-bbf4-8c72c30df5be")
+
+#
+# # Depot Changes
+#
+@testset "Depot setup" begin
+    isolate() do
+        # Lets make sure we start with a clean slate.
+        rm(LOADED_DEPOT; force=true, recursive=true)
+        mkdir(LOADED_DEPOT)
+        # And set the loaded depot as our working depot.
+        empty!(DEPOT_PATH)
+        push!(DEPOT_PATH, LOADED_DEPOT)
+        # Now we double check we have a clean slate.
+        @test isempty(Pkg.dependencies())
+        # A simple `add` should set up some things for us:
+        Pkg.add(Pkg.PackageSpec(;name="Example", version="0.5.3"))
+        # - `General` should be initiated by default.
+        regs = Pkg.Registry.status(;as_api=true)
+        @test length(regs) == 1
+        reg = regs[1]
+        @test reg.name == "General"
+        @test reg.uuid == general_uuid
+        # - The package should be installed correctly.
```

Comment:
This only checks if the project + manifest files were correctly updated. Don't we want to use the `isinstalled` here to see that the files are actually downloaded?

Reply:
Oh, that's the `isdir(pkg.source)`?